src/test/java/BaseTest.java

```diff
@@ -1,11 +1,269 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
-import org.testng.annotations.BeforeSuite;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.*;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import java.net.MalformedURLException;
+import java.net.URI;
+import org.openqa.selenium.Keys;
+import java.net.URL;
+import java.util.HashMap;
+
+import javax.swing.*;
+import java.time.Duration;
 
 public class BaseTest {
+    public static WebDriver driver = null;
+
+//    private WebDriver driver = null;
+
+    public static String url = "https://qa.koel.app/";
+
+    public static WebDriverWait wait = null;
+
+    public static Actions actions = null;
+
+    private static final ThreadLocal<WebDriver> threadDriver = new ThreadLocal<>();
+
 
     @BeforeSuite
-    static void setupClass() {
-        WebDriverManager.chromedriver().setup();
+//    public static void chromeConfigs() {
+//
+//        if (System.getProperty("os.name").toLowerCase().contains("ios")) {
+//            System.setProperty("webdriver.chrome.driver", "chromedriver");
+//        }
+//    }
+//    static void setupClass() {
+//
+//        WebDriverManager.chromedriver().setup();
+//        WebDriverManager.FirefoxDriver().setup();
+//
+//
+////    }
```

Comment:
I recommend not using code comments: for, example, when you have a code that cannot be understood, then comments are not the solution. Try to rewrite the code or rename fields and other elements in a way that the reader will understand the action that you are doing there.